src/frontend/app/features/service-catalog/add-service-instance/specify-details-step/specify-details-step.component.ts

```diff
@@ -0,0 +1,242 @@
+import { PaginationMonitorFactory } from '../../../../shared/monitors/pagination-monitor.factory';
+import {
+  SetCreateServiceInstance,
+  SetOrg,
+  SetServiceInstanceGuid,
+  SetSpace,
+} from '../../../../store/actions/create-service-instance.actions';
+import { CreateServiceInstance, GetServiceInstances } from '../../../../store/actions/service-instances.actions';
+import { AppState } from '../../../../store/app-state';
+import { entityFactory, organizationSchemaKey, serviceInstancesSchemaKey } from '../../../../store/helpers/entity-factory';
+import { createEntityRelationPaginationKey } from '../../../../store/helpers/entity-relations.types';
+import { RequestInfoState } from '../../../../store/reducers/api-request-reducer/types';
+import { getPaginationObservables } from '../../../../store/reducers/pagination-reducer/pagination-reducer.helper';
+import { selectRequestInfo } from '../../../../store/selectors/api.selectors';
+import { selectOrgGuid, selectServicePlan } from '../../../../store/selectors/create-service-instance.selectors';
+import { APIResource } from '../../../../store/types/api.types';
+import { CloudFoundryEndpointService } from '../../../cloud-foundry/services/cloud-foundry-endpoint.service';
+import { ServicesService } from '../../services.service';
+
+@Component({
+  selector: 'app-specify-details-step',
+  templateUrl: './specify-details-step.component.html',
+  styleUrls: ['./specify-details-step.component.scss'],
+})
+export class SpecifyDetailsStepComponent implements OnInit, OnDestroy, AfterContentInit {
+
+  stepperForm: FormGroup;
+  serviceInstanceNameSub: Subscription;
+  allServiceInstances$: Observable<APIResource<IServiceInstance>[]>;
+  validate: Observable<boolean>;
+  orgSubscription: Subscription;
+  spaceSubscription: Subscription;
+  allServiceInstanceNames: string[];
+  tagsVisible = true;
+  tagsSelectable = true;
+  tagsRemovable = true;
+  tagsAddOnBlur = true;
+  separatorKeysCodes = [ENTER, COMMA, SPACE];
+  tags = [];
+
+  spaces$: Observable<APIResource<ISpace>[]>;
+  orgs$: Observable<APIResource<IOrganization>[]>;
+
+  nameTakenValidator = (): ValidatorFn => {
+    return (formField: AbstractControl): { [key: string]: any } =>
+      !this.checkName(formField.value) ? { 'nameTaken': { value: formField.value } } : null;
+  }
+
+  constructor(
+    private store: Store<AppState>,
+    private servicesService: ServicesService,
+    private paginationMonitorFactory: PaginationMonitorFactory,
+    private snackBar: MatSnackBar,
+  ) {
+
+    this.stepperForm = new FormGroup({
+      name: new FormControl('', [Validators.required, this.nameTakenValidator()]),
+      org: new FormControl('', Validators.required),
+      space: new FormControl('', Validators.required),
+      params: new FormControl(''),
+      tags: new FormControl(''),
+    });
+
+    const getAllOrgsAction = CloudFoundryEndpointService.createGetAllOrganizations(servicesService.cfGuid);
```

Comment:
It's worth bearing in mind that this is a monster of an action that will fetch a lot of data. It might be worth using the GetAllOrganizations with just the org-space relationship